README.md

```diff
@@ -4,3 +4,186 @@
+
+* Entities
+* Components
+* Systems. 
+
+In our implementation, we use the type `World` to work with those `System`s. Each `System` can have references to any number (including 0) of Entities. And each `Entity` can have as many `Component`s as desired. 
+
+An example of creating a `World`, adding a `System` to it, and update all systems
+```go
+// Declare the world - you can also use "var world ecs.World" 
+world := ecs.World{}
+
+// You can add as many Systems here as you like. The RenderSystem provided by `engo` is just an example. 
+world.AddSystem(&engo.RenderSystem{})
+
+// This will usually be called within the game-loop, in order to update all Systems on every frame.  
+world.Update(0.125) // 0.125 would be the time in seconds since the last update
+```
+
+## System
+We've been talking about `System`s, but what are they? Anything that implements the interface, can be used as a `System`:
+
+```go
+type System interface {
+	// Update is ran every frame, with `dt` being the time in seconds since the last frame
+	Update(dt float32)
+
+	// Delete should remove the entity from the system completely
+	Remove(e BasicEntity)
+}
+```
+
+What does this say? It needs to have an `Update` method (which is called from `world.Update`), and it needs to have a `Remove(ecs.BasicEntity)` method. Why require a Remove method, but not an Add method? Because there's no 'generic' `Add` method (the parameters may change), while in order to remove something, all you need it the unique identifier (as provided by the `BasicEntity`). 
+
+### Initialization
+Optionally, your `System` may implement the `Initializer` interface, which allows you to do initialization for the given `World`. Basically, it allows you to initialize values, without having to call the function manually before adding it to the `World`. Whenever you add a `System` (one that implements the `Initializer` interface) to the world, the `New` method will be called. 
+
+```go
+type Initializer interface {
+	// New is the initialisation of the System, and may be used to initialize some values beforehand, like storing
+	// a reference to the World
+	New(*World)
+}
+```
+
+### Priority
+Optionally, your `System` may implement the `Prioritizer` interface, which allows the `World` to sort the `System`s based on that priority. If omitted, a value of `0` is assumed. 
+
+```go
+type Prioritizer interface {
+	// Priority indicates the order in which Systems should be executed per iteration, higher meaning sooner. Default is 0
+	Priority() int
+}
+```
+
+## Entities and Components
+Where do the entities come in? `System`s can "use" entities by manipulating values of the `Component`s of those entities. An `Entity` is no more than a wrapper which combines multiple `Component`s and adds a unique identifier to the whole. Because the precise definition of those `Component`s can vary, `ecs` provides no `Component`s. The `engo` package offers lots of `Component`s and `System`s to work with, out of the box. 
+
```

Comment:
For newcomers to the ECS paradigm, we may wish to stress that *all* logic (i.e. code) should happen within `System`s. `Component`s store data (which is used by the Systems), but do nothing else. They have no methods or no code attached to them. The description of entities is great! If we want to make it abundantly clear, we could mention that the unique identifier is often simple an incrementing integer value, thus nothing *magic*. Anything making ECS feel less magic may help newcomers feel less intimidated.